### ISSUE TITLE
RN issue: Cannot choose between the following variants of com.facebook.react:react-native

### ISSUE THREAD

**Issue by esalaza** [maxkomarychev/react-native-ultimate-config#115]
# Describe the bug

Please see https://github.com/facebook/react-native/issues/35210 for the full explanation. ~~The suggested fix in this case would be to bump the `react-native` dependency to: 0.64.4 (https://github.com/facebook/react-native/releases/tag/v0.64.4)~~

# Steps to Reproduce

```
nvm install lts/hydrogen
nvm use lts/hydrogen
npm uninstall -g react-native-cli @react-native-community/cli
npx react-native init AwesomeProject
cd AwesomeProject
npx react-native start
npx react-native run-android
```
The app should launch without issues. Stop Metro. Kill the App. Then

```
npm i react-native-ultimate-config
npx react-native start
npx react-native run-android
```

And you should get something like

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/.../AwesomeProject/node_modules/react-native-ultimate-config/android/build.gradle' line: 113

* What went wrong:
A problem occurred configuring project ':react-native-ultimate-config'.
> Could not resolve all files for configuration ':react-native-ultimate-config:implementation'.
   > Could not resolve com.facebook.react:react-native:+.
     Required by:
         project :react-native-ultimate-config
      > Cannot choose between the following variants of com.facebook.react:react-native:0.70.5:
          - debugVariantDefaultRuntimePublication
          - releaseVariantDefaultRuntimePublication
        All of them match the consumer attributes:
          - Variant 'debugVariantDefaultRuntimePublication' capability com.facebook.react:react-native:0.70.5:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.dependency.bundling 'external' but the consumer didn't ask for it
                  - Provides org.gradle.libraryelements 'aar' but the consumer didn't ask for it
                  - Provides org.gradle.status 'release' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
          - Variant 'releaseVariantDefaultRuntimePublication' capability com.facebook.react:react-native:0.70.5:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.dependency.bundling 'external' but the consumer didn't ask for it
                  - Provides org.gradle.libraryelements 'aar' but the consumer didn't ask for it
                  - Provides org.gradle.status 'release' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
```

# Expected behavior

The app should continue running without issues.

## OS

- [ ] ios
- [x] android

Thank you for this awesome tool :)

**Comment by esalaza** [comment]
This is actually a duplicate of #107.

**Comment by JeremyBradshaw7** [comment]
Is this issue sporadic? I did a build successfully earlier today, but now cannot get past this error.

**Comment by marcshilling** [comment]
@esalaza are you sure? This is a different error than in 107. I'm getting this on RN 0.69.7, which is the version that the RN team patched to resolve the upstream issue.

**Comment by maxkomarychev** [comment]
There was a massive refactoring made to the library and this problem could have been fixed in a new version 4.1.0-alpha.0 (https://github.com/maxkomarychev/react-native-ultimate-config/pull/128)

Please give it a try!! Thank you!
